parquet-hadoop/src/test/java/parquet/hadoop/TestInputFormat.java

```diff
@@ -378,6 +387,62 @@ public void testDeprecatedConstructorOfParquetInputSplit() throws Exception {
     shouldSplitStartBe(splits, 0, 50);
   }
 
+  @Test
+  public void testGetFootersReturnsInCorrectOrder() throws IOException {
+    File tempDir = Files.createTempDir();
+    tempDir.deleteOnExit();
+    int numFiles = 10; // create a nontrivial number of files so that it actually tests getFooters() returns files in the correct order
+
+    String url = "";
+    for (int i = 0; i < numFiles; i++) {
+      File file = new File(tempDir, String.format("part-%05d.parquet", i));
+      createParquetFile(file);
+      if (i > 0) {
+        url += ",";
+      }
+      url += "file:" + file.getAbsolutePath();
+    }
+
+    Job job = new Job();
+    FileInputFormat.setInputPaths(job, url);
+    List<Footer> footers = new ParquetInputFormat<Object>().getFooters(job);
+    for (int i = 0; i < numFiles; i++) {
+      Footer footer = footers.get(i);
+      File file = new File(tempDir, String.format("part-%05d.parquet", i));
+      assertEquals("file:" + file.getAbsolutePath(), footer.getFile().toString());
+    }
+  }
+
+  private void createParquetFile(File file) throws IOException {
```

Comment:
This was my first attempt at creating a test parquet file. Please let me know if there's an easier way.